pennylane/data/attributes/operator/operator.py

```diff
@@ -214,7 +215,7 @@ def _ops_to_hdf5(
             op_key = f"op_{i}"
             if isinstance(op, (qml.ops.Prod, qml.ops.SProd, qml.ops.Sum)):
                 op = op.simplify()
-            if type(op) not in self.consumes_types():
+            if type(op) not in self.supported_ops():
```

Comment:
Do we still need to maintain this list of supported ops? Or can we now say "it's an operator, therefore we can serialize it"?

Reply:
The `DatasetOperator` class doesn't use Pytrees, so it still has the same limitations. New datasets will use the `DatasetPyTree` type to serialize operators, but we need to keep the `DatasetOperator` around so that existing datasets will still work.

There should probably be some kind of plan to deprecate the old `DatasetOperator` class for removal in a future version. This will require converting existing datasets to use the pytree attribute type.